salt/states/boto3_route53.py

```diff
@@ -798,3 +826,204 @@ def rr_absent(name, HostedZoneId=None, DomainName=None, PrivateZone=False,
+        'EnableSNI', 'InsufficientDataHealthStatus'
+    ]
+    resettables = [ 'FullyQualifiedDomainName', 'Regions', 'ResourcePath', 'ChildHealthChecks' ]
+
+    checks = __salt__['boto3_route53.find_health_checks'](Name=Name, region=region, key=key, keyid=keyid, profile=profile)
+
+    if len(checks) > 1:
+        ret['comment'] = 'Given criteria matched more than one health check.'
+        log.error(ret['comment'])
+        ret['result'] = False
+        return ret
+    elif len(checks) == 0:
+        create = True
+        CallerReference = CallerReference if CallerReference else str(uuid.uuid4())  # future lint: disable=blacklisted-function
+    else:
+        check = checks[0].get('HealthCheckConfig')
+        for u in updatables:
+            if u in [ 'HealthCheckVersion' ]:
+                continue
+            if _ensure_unicode(params.get(u)) != _ensure_unicode(check.get(u)):
+                update = True
+                break
+            if ((ChildHealthChecks is not None) and (check.get('ChildHealthChecks') is not None)):
+                if sorted(ChildHealthChecks) != sorted(check.get('ChildHealthChecks')):
+                    update = True
+            if ((Regions is not None) and (check.get('Regions') is not None)):
+                if sorted(Regions) != sorted(check.get('Regions')):
+                    update = True
+            if ((AlarmIdentifier is not None) and (check.get('AlarmIdentifier') is not None)):
+                if AlarmIdentifier.get('Name') != check.get('AlarmIdentifier', {}).get('Name'):
+                    update = True
+                if AlarmIdentifier.get('Region') != check.get('AlarmIdentifier', {}).get('Region'):
+                    update = True
+        for key in resettables:
+            if params.get(key) is None and check.get(key) is not None:
+                update = True
+                reset_elements.append(key)
+
+    if not create and not update:
+        ret['comment'] = 'Route 53 health check {} is already in the desired state.'.format(Name)
+        log.info(ret['comment'])
+        return ret
+    elif create:
+        keys = [
+            'IPAddress', 'Port', 'Type', 'ResourcePath', 'FullyQualifiedDomainName', 'SearchString',
+            'RequestInterval', 'FailureThreshold', 'MeasureLatency', 'Inverted', 'Disabled', 'HealthThreshold',
+            'EnableSNI', 'InsufficientDataHealthStatus', 'Regions', 'AlarmIdentifier', 'ChildHealthChecks', 'CallerReference'
+        ]
+
+        items = {key: params.get(key) for key in keys if params.get(key) is not None}
+        args = {'Name': Name, 'region': region, 'key': key, 'keyid': keyid, 'profile': profile}
+        args.update(items)
+        log.debug(' ==> {}'.format(items))
+
+        if __opts__['test']:
+            ret['comment'] = 'Route 53 health check {} would be created.'.format(Name)
+            ret['changes'] = items
+            ret['result'] = None
```

Comment:
I seem to recall another state where it was returning `None` for the result instead of `True` during a test run. I could be wrong about that part, perhaps it was just `changes`, but I do know that whatever it was doing led to an erroneous reporting (that is, instead of the test saying, yes, something would have changed, it just *said* there were changes but everything was already in the correct state).

If that doesn't seem correct, I can run it down and make sure that what I'm thinking of is correct.

Reply:
This is taken from anothe boto module, but it looks like it is present everywhere:

$ grep -F "ret['result'] = None" salt/states/*|wc -l
     971